pennylane/ops/channel.py

```diff
@@ -224,11 +224,63 @@ def _kraus_matrices(cls, *params):
         return [K0, K1, K2, K3]
 
 
+class QubitChannel(Channel):
+    r"""QubitChannel(K_list, wires)
+    Apply an arbitrary fixed quantum channel.
+
+    Kraus matrices that represent the fixed channel are provided
+    as a list of NumPy arrays.
+
+    **Details:**
+
+    * Number of wires: Any (the operation can act on any number of wires)
+    * Number of parameters: 1
+    * Gradient recipe: None
+
+    Args:
+        K_list (list[array[complex]]): List of Kraus matrices
+        wires (Union[Wires, Sequence[int], or int]): the wire(s) the operation acts on
+    """
+    num_params = 1
+    num_wires = AnyWires
+    par_domain = "L"
+    grad_method = None
+
+    @classmethod
+    def _kraus_matrices(cls, *params):
+        K_list = params[0]
+
+        # check all Kraus matrices are square matrices
+        if not all(K.shape[0] == K.shape[1] for K in K_list):
+            raise ValueError(
+                "Only channels with similar input and output Hilbert space dimensions can be applied."
+            )
+
+        # check all Kraus matrices have the same shape
+        if not all(K.shape == K_list[0].shape for K in K_list):
+            raise ValueError("All Kraus matrices must have the same shape.")
+
+        # check the dimension of all Kraus matrices are valid
+        if not all(K.ndim == 2 for K in K_list):
+            raise ValueError(
+                "Dimension of all Kraus matrices must be (2**num_wires, 2**num_wires)."
+            )
+
+        # check that the channel represents a trace-preserving map
+        K_arr = np.array(K_list)
+        Kraus_sum = np.einsum("ajk,ajl->kl", K_arr.conj(), K_arr)
+        if not np.allclose(Kraus_sum, np.eye(K_list[0].shape[0])):
+            raise ValueError("Only trace preserving channels can be applied.")
+
+        return K_list
```

Comment:
These checks are great, but they seem pretty heavy. From my understanding of how things are working, isn't `_kraus_matrices` called quite regularly? Would it make sense to move these checks to the `__init__` of `QubitChannel`?

Reply:
Hm, I don't see how that would be any different.

Reply:
If we place these checks in `QubitChannel.__init__()`, they would just be called once when the channel is instantiated (e.g., a similar thing happens for `PauliRot`)

Instead, they are run in `_kraus_matrices()`, which I believe is called every time the Kraus matrices are requested by the device. This doesn't seem necessary since the kraus matrices are static and only set upon instantiation.

Reply:
hm nice! I have incorporated this suggestion :)